src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,120 @@
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        String query = "INSERT INTO products (name, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            Product newProduct = new Product(product.getName(), product.getPrice());
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                newProduct.setId(resultSet.getLong(1));
+            }
+            return newProduct;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create product", e);
+        }
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET name = ?, "
+                + "price = ? WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.setLong(3, product.getId());
+            int successfulUpdate = statement.executeUpdate();
+            if (successfulUpdate != 0) {
+                return product;
+            }
+            throw new SQLException();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update product", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long productId) {
+        String query = "SELECT * FROM products WHERE product_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, productId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Product product = getProductFromResultSet(resultSet);
+                return Optional.of(product);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get product by id", e);
```

Comment:
let's make message more informative (add some info about product, that wasn't retrieved)